types/index.d.ts

```diff
@@ -0,0 +1,487 @@
+    readonly schema: Schema<Registry>;
+
+    /** Creates a model of the given type. */
+    readonly create: Schema<Registry>["create"];
+
+    /** Whether or not Mirage should log all requests/response cycles. */
+    logging: boolean;
+
+    /** A default number of ms to artificially delay responses for all routes. */
+    timing: number;
+
+    /** A default prefix applied to all subsequent route definitions. */
+    namespace: string;
+
+    /** Sets a string to prefix all route handler URLs with. */
+    urlPrefix: string;
+
+    /** Actual Pretender instance */
+    pretender: PretenderServer;
+
+    /** Creates multiple models of the given type. */
+    createList<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<Init>
+    >(modelName: K, count: number, data?: Data): Array<Init & Data>;
+
+    /** Handle a GET request to the given path. */
+    get(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a POST request to the given path. */
+    post(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PUT request to the given path. */
+    put(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PATCH request to the given path. */
+    patch(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle an OPTIONS request to the given path. */
+    options(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a DELETE request to the given path. */
+    del(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    delete(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Pass through one or more URLs to make real requests. */
+    passthrough(urls?: ((request: Request) => any) | string | string[]): void;
+
+    /** Load all available fixture data matching the given name(s). */
+    loadFixtures(...names: string[]): void;
+
+    seeds(server: Server): void;
+
+    routes(): void;
+
+    /** Shutdown the server and stop intercepting network requests. */
+    shutdown(): void;
+  }
+}
+
+declare module "miragejs/db" {
+  import DbCollection from "miragejs/db-collection";
+  import IdentityManager from "miragejs/identity-manager";
+
+  /** The in-memory database containing all currently active data keyed by collection name. */
+  export default class Db {
+    [key: string]: DbCollection | any;
+
+    constructor(initialData: [], identityManagers?: IdentityManager[]);
+
+    createCollection(name: string, initialData?: any[]): void;
+    dump(): void;
+    emptyData(): void;
+    loadData(data: any): void;
+  }
```

Comment:
This ends up pretty gnarly, but it does the right thing by merging two distinct types as the type exported for `Db`:

```suggestion
  type DbLookup = {
    [key: string]: DbCollection;
  };

  class DbClass {
    constructor(initialData: [], identityManagers?: IdentityManager[]);

    createCollection(name: string, initialData?: any[]): void;
    dump(): void;
    emptyData(): void;
    loadData(data: any): void;
  }

  /** The in-memory database containing all currently active data keyed by collection name. */
  export const Db: DbClass;
  export type Db = DbClass & DbLookup;
  export default Db;
```